### PR TITLE
fix: filter on the block unique classname for saving styles

### PR DESCRIPTION
**Pull request by decodekult**
Add a way to match the modified block unique class on styles and on HTML nodes. Also, add a note on a filter since the original value will always be empty - please consider deprecating the filter altogether.

Ref: https://github.com/gambitph/Stackable/issues/2588

### CODE REVIEW

File: src/components/block-css/index.js [hunk 1]
```diff
@@ -345,6 +345,10 @@ const BlockCss = props => {
 		collapsedSelector = collapsedSelector.replace( /[^^?](.%s)([^-])/g, `$1-${ instanceId }$2` )
 	}
 
+	if ( ! props.editorMode ) {
+		blockUniqueClassName = applyFilters( 'stackable.block-css.uniqueClass.save', blockUniqueClassName, attributes )
+	}
+
```
Comment (bfintal) [inline src/components/block-css/index.js:351]: Looks good to me!

I was thinking that it also might be a good idea to also do this for the `edit` part just for the sake of completion:

```suggestion
	blockUniqueClassName = applyFilters( `stackable.block-css.uniqueClass.${ props.editorMode ? 'edit' : 'save' }`, blockUniqueClassName, attributes )

```

Comment (decodekult) [reply src/components/block-css/index.js]: Hi Benjamin!

Yes, I noticed that there is some asymmetry between `save` and `edit` processes. I decided not to move to the edit part for some reasons:
- All the `edit`-related hooks introduced to modify unique class names only pass the filtering `className` value, never the `attributes`, since the set of attributes being passed to components executing those filters is always a subset of the block attributes, as stated [here](https://github.com/gambitph/Stackable/pull/2446#issuecomment-1370968384). If we want to have a `save` and an `edit` set of filters, I would suggest following the same schema and pass `attributes` only to the `save` version, and in fact I am pushing this change to the current pull request.
- The `edit` flow it a little more entangled. It passes over the `BlockCssEdit` component. In it, we gather the `instanceId` from the WordPress native query loop block, matching the ID of the current post (I assume), and we pass that `instanceId` to the `BlockCss` component. With that `instanceId`, the `BlockCss` component composes one unique classname by appending it to the result of a `getBlockUniqueClassname` call (note that the `BlockCss` component itself has some commented-out code to gather the `instanceId` by itself). In the end, the `BlockCss` composes the CSS output for the `edit` flow by generating an unique classname counting on the `instanceId`. **But** the `BlockCssEdit` component also applies the `stackable.block-styles.edit` filter, and passes to it the generated CSS for the editing block, and also a block unique classname that **does not include** any reference to the `instanceId` in case it exists.

In short, I feel that there is also a mismatch between unique classnames on the edit process, especially when the block is edited in the WordPress new site editor and especially when dealing with the current post context when inside a query loop, but I could not reproduce any issue related to this, so I decided to skip it and focus on the saving mechanism.

Let me know if you would like me to open that party can :-) I would rather investigate slightly deeper and open a new issue if I find anything.

**Comment by bfintal** [comment]
@decodekult good points, this new change looks good to me already :)
